### PR TITLE
Independent Config Route Creation for Packages & Modules

### PR DESCRIPTION
**Pull request by Michael-Vol**
This PR adds support for generating config routes independently for each package and module. It also introduces strict convict configuration validation in order to prevent users from adding irrelevant fields into the config.

**What kind of change does this PR introduce?** 

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other (please describe)

**Does this PR introduce a breaking change?** 

- [ ] Yes
- [x] No

**The PR fulfills these requirements:**

- [x] It's submitted to the `main` branch
- [ ] When resolving a specific issue, it's referenced in the PR's description  (e.g. `fix #xxx`, where "xxx" is the issue number)

### CODE REVIEW

File: packages/core/src/routes/index.ts [hunk 1]
```diff
@@ -47,6 +55,47 @@ export class HttpServer {
         }
       },
     );
+
+    // this._routingManager.route(
```
Comment (kon14) [inline packages/core/src/routes/index.ts:59]: Leftovers

File: packages/core/src/routes/index.ts [hunk 2]
```diff
@@ -1,16 +1,24 @@
 import ConduitGrpcSdk, {
+  ConduitError,
```
Comment (kon14) [inline packages/core/src/routes/index.ts:2]: Remove these unused types.

File: packages/core/src/config-manager/index.ts [hunk 3]
```diff
@@ -275,22 +312,9 @@ export default class ConfigManager implements IConfigManager {
...
-    this.sdk
-      .getAdmin()
-      .registerRoute(
-        adminRoutes.getPatchConfigRoute(
-          this.grpcSdk,
-          this.sdk,
-          this.serviceDiscovery.registeredModules,
-        ),
-      );
+    // this.sdk.getAdmin().registerRoute(adminRoutes.getGetConfigRoute(this.grpcSdk));
```
Comment (kon14) [inline packages/core/src/config-manager/index.ts:315]: Leftovers.

File: packages/core/src/config-manager/admin/routes/GetConfig.route.ts [hunk 4]
```diff
@@ -6,14 +6,9 @@ import ConduitGrpcSdk, {
...
 } from '@conduitplatform/grpc-sdk';
-import { isNil } from 'lodash';
-import * as models from '../../models';
 import { ConduitRoute, ConduitRouteReturnDefinition } from '@conduitplatform/hermes';
 
-export function getGetConfigRoute(
-  grpcSdk: ConduitGrpcSdk,
-  registeredModules: Map<string, RegisteredModule>,
-) {
+export function getGetConfigRoute(grpcSdk: ConduitGrpcSdk) {
```
Comment (kon14) [inline packages/core/src/config-manager/admin/routes/GetConfig.route.ts:11]: Remove this route handler file.

File: packages/core/src/config-manager/admin/routes/PatchConfig.route.ts [hunk 5]
```diff
@@ -8,11 +8,7 @@ import ConduitGrpcSdk, {
 } from '@conduitplatform/grpc-sdk';
 import { ConduitRoute, ConduitRouteReturnDefinition } from '@conduitplatform/hermes';
 
-export function getPatchConfigRoute(
-  grpcSdk: ConduitGrpcSdk,
-  conduit: ConduitCommons,
-  registeredModules: Map<string, RegisteredModule>,
-) {
+export function getPatchConfigRoute(grpcSdk: ConduitGrpcSdk, conduit: ConduitCommons) {
```
Comment (kon14) [inline packages/core/src/config-manager/admin/routes/PatchConfig.route.ts:11]: Remove this route handler file.

File: libraries/grpc-sdk/src/classes/ModuleManager.ts [hunk 6]
```diff
@@ -67,10 +80,12 @@ export class ModuleManager<T> {
   private async postRegisterLifecycle(): Promise<void> {
     await this.module.onRegister();
     if (this.module.config) {
+      let configSchema = this.module.config.getSchema();
```
Comment (kon14) [inline libraries/grpc-sdk/src/classes/ModuleManager.ts:83]: Make this const.

**Review by kon14 — APPROVED** [review]

**Review by kon14 — CHANGES_REQUESTED** [review]
Approved this accidentally.

**Review by kon14 — APPROVED** [review]

**Review by kkopanidis — APPROVED** [review]

**Pull request merged**